src/java.base/share/classes/java/lang/String.java

```diff
@@ -1443,7 +1443,7 @@ public String(byte[] bytes) {
      *         A {@code StringBuffer}
      */
     public String(StringBuffer buffer) {
-        this(buffer.toString());
+        this(buffer, null);
```

Comment:
This method is no longer synchronized on StringBuffer

Reply:
you're right. fixed in revision#2. 
this could be very tricky to discover. thanks for the head-up!